### PR TITLE
[SPARK-23700][PYTHON] Cleanup imports in pyspark.sql

### PR DESCRIPTION
**Pull request by BryanCutler**
## What changes were proposed in this pull request?

This cleans up unused imports, mainly from pyspark.sql module.  Added a note in function.py that imports `UserDefinedFunction` only to maintain backwards compatibility for using `from pyspark.sql.function import UserDefinedFunction`. 

## How was this patch tested?

Existing tests and built docs.

### CODE REVIEW

**Comment by BryanCutler** [comment]
@HyukjinKwon and @ueshin I tried to be pretty conservative and only remove imports that were obviously not being used.

File: python/pyspark/sql/functions.py [hunk 1]
```diff
@@ -28,10 +27,10 @@
 
 from pyspark import since, SparkContext
 from pyspark.rdd import ignore_unicode_prefix, PythonEvalType
-from pyspark.serializers import PickleSerializer, AutoBatchedSerializer
 from pyspark.sql.column import Column, _to_java_column, _to_seq
 from pyspark.sql.dataframe import DataFrame
 from pyspark.sql.types import StringType, DataType
+# Keep UserDefinedFunction import for backwards compatible import; moved in SPARK-22409
 from pyspark.sql.udf import UserDefinedFunction, _create_udf
```
Comment (BryanCutler) [inline python/pyspark/sql/functions.py]: Not sure if there is a better way to do this other than importing `UserDefinedFunction` here, but hopefully the note will show the intent.

**Comment by BryanCutler** [comment]
Jenkins retest this please

File: python/pyspark/sql/functions.py (hunk 1 above)
Comment (HyukjinKwon) [reply python/pyspark/sql/functions.py]: yea, I think we should better keep this import and the comment looks good.

**Review by felixcheung — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]
LGTM

**Comment by BryanCutler** [comment]
Thanks @felixcheung and @HyukjinKwon !

**Pull request closed without merging**